### PR TITLE
[iOS] Use react native 0.62 hook for dark mode

### PR DESCRIPTION
**Pull request by marecar3**
## Description
With React Native 0.62 we got built-in dark mode feature: https://reactnative.dev/blog/2020/03/26/version-0.62#new-dark-mode-features
which allows us to remove `react-native-dark-mode` library from the project.

Gutenberg mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2120

## How has this been tested?

The dark mode should work as expected both in the Demo app and the WPiOS app.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by gziolo — COMMENTED** [review]
Nice 👍

File: packages/compose/src/hooks/use-preferred-color-scheme/index.ios.js [hunk 1]
```diff
@@ -1,18 +1,13 @@
...
 /**
  * Returns the color scheme value when it changes. Possible values: [ 'light', 'dark' ]
  *
  * @return {string} return current color scheme.
  */
-const usePreferredColorScheme = useDarkModeContext;
+const usePreferredColorScheme = useColorScheme;
 
-export default usePreferredColorScheme;
+export default usePreferredColorScheme;
```
Comment (etoledom) [inline packages/compose/src/hooks/use-preferred-color-scheme/index.ios.js:13]: There seems to be a missing empty line at the end of the file.

**Review by etoledom — APPROVED** [review]
Thank you @marecar3 for this!
It's working great 🎉 

Let's get rid of the external dependency 😁

File: packages/compose/src/hooks/use-preferred-color-scheme/index.ios.js (hunk 1 above)
Comment (marecar3) [reply packages/compose/src/hooks/use-preferred-color-scheme/index.ios.js]: Fixed. Thanks!

**Comment by marecar3** [comment]
> Thank you @marecar3 for this!
> It's working great 🎉
> 
> Let's get rid of the external dependency 😁

Thanks for reviewing this @etoledom!

**Pull request merged**